docs/BUILD.md

```diff
@@ -0,0 +1,449 @@
+* [linuxkit](https://github.com/linuxkit/linuxkit) - CLI used to build actual EVE bootable images.
+* [dep](https://github.com/golang/dep) - CLI used to install dependencies for other "Installed as Needed" requirements
+
+Each "Installed as Needed" build tool will place its final executable in `build-tools/bin/`, e.g. `build-tools/bin/linuxkit` or `build-tools/bin/manifest-tool`. Each tool, in turn, has a directory in `build-tools/src/`, e.g. `build-tools/src/linuxkit/`. The `src/` directory does _not_ contain all of the code for the tool. Instead, it has the following:
+
+* `Gopkg.toml` - with all of the dependencies including the specific version of the package itself as a `constraint`
+* `Gopkg.lock` - generated from a run
+* `dummy.go` - an empty file to give go something to build
+
+To build a specific tool, you can execute `make bin/<tool-name>`, e.g. `make bin/manifest-tool`, in `build-tools`, e.g.:
+
+```
+make -C build-tools bin/manifest-tool
+```
+
+This target does the following:
+
+1. Set `GOPATH=$PWD`, i.e. the `build-tools/` directory
+2. If `dep` isn't available, `go get` it
+3. `cd src/<tool> && dep ensure -v`. This step installs the tool's go source in `vendor/`, as it is in `Gopkg.toml`
+4. `cd src/<tool>/vendor/<path>` ; this path will change based on the tool.
+5. `go build <options> -o $GOPATH/bin/<tool>`
+
+To build all of the tools, run `make -C build-tools all`, or in the project root directory, just `make build-tools`
+
+#### Reasoning
+
+The normal process for installing go-based binaries is to execute `go get` with options, e.g.
+
+```
+go get -u github.com/linuxkit/linuxkit/src/cmd/linuxkit
+go get -u github.com/estesp/manifest-tool
+```
+
+EVE uses a somewhat non-standard build process for these tools to ensure specific versions without polluting the user's normal workspace. 
+
+## Output Components
+
+The following are the output components from the build process and their purpose. There are two kinds of components: final, intended for actual direct usage, and interim, used to build the final components. Some interim components may be removed as part of the build finalization process.
+
+#### Final
+
+* `fallback.img` - a symlink to `fallback.qcow2`
+* `fallback.qcow2` - the final bootable live disk image in [qcow2](https://en.wikipedia.org/wiki/Qcow) format
+* `installer.raw` - a bootable image that can install EVE on a local device. The installer is intended to be flashed to a USB or SD device, or booted via PXE, and then run to install on a local drive.
+* `installer.iso` - a bootable ISO image with a hidden EFI boot partition and an installer partition, with the contents of `installer.raw`. The installer is intended to be booted in a manner typical of iso files, and then run to install on a local drive.
+
+#### Interim
+
+* `rootfs.img` - a live bootable rootfs filesystem image. This can be either [squashfs](https://en.wikipedia.org/wiki/SquashFS) (default) or [ext4](https://en.wikipedia.org/wiki/Ext4). 
```

Comment:
This can be used to update the baseimage using the API.

Reply:
I don't understand. Do you mean that a built newer version of `rootfs.img`, in addition to being directly flashed or installed via `installer.img`, also can be distributed via the API and our a/b upgrade procedure?

Reply:
Correct. The format for the baseos images used for updates is the qcow2 format in rootfs.img
(this is how we do it today; upload the rootfs and then use the API to point the device where it should download it from)